src/extension/wps/wps-core/src/main/java/org/geoserver/wps/executor/WPSExecutionManager.java

```diff
@@ -391,6 +391,13 @@ private ExecuteResponseType execute() {
                 SubProgressListener executionListener = new SubProgressListener(listener,
                         inputPercentage, executionPercentage);
                 notifier.checkDismissed();
+                
+                //The system only moves from Queued to Running when the progress notifier fires (!!!).  This fires when inputs are read.
+                //   However, if there are no inputs, it never moves to running state!
+                //Since we are executing, we can move this from queued to running state. 
+                if (status.getPhase() == ProcessState.QUEUED) {
```

Comment:
A pure wrapper is cleaner, we don't end up needing extra parameters, it just needs to send the right event on the listener before delegating the execution call, and it's independent on how the process manager is implemented (there are custom ones out there)